### ISSUE TITLE
OnActiveSec is the wrong timer property for triggering the service unit on an interval

### ISSUE THREAD

**Issue by nathwill** [chef-cookbooks/chef-client#613]
### Cookbook version
11.0.5

### Chef-client version
14.9.13

### Platform Details
CentOS 7.6.1810

### Scenario:
systemd timer unit triggered activation of chef-client.service

### Steps to Reproduce:
set up the timer/service units as the cookbook currently does 

e.g.
```
PRODUCTION [root@bastion.prod.iad01.treehouse chef-client]$ cat /etc/systemd/system/chef-client.timer
[Unit]
Description = chef-client periodic run

[Install]
WantedBy = timers.target

[Timer]
OnBootSec = 1min
OnActiveSec = 1800sec
RandomizedDelaySec = 900sec
PRODUCTION [root@bastion.prod.iad01.treehouse chef-client]$ cat /etc/systemd/system/chef-client.service 
[Unit]
Description = Chef Client daemon
After = network.target auditd.service

[Service]
Type = oneshot
EnvironmentFile = /etc/sysconfig/chef-client
ExecStart = /usr/bin/chef-client -c $CONFIG $OPTIONS
ExecReload = /bin/kill -HUP $MAINPID
SuccessExitStatus = 3

[Install]
WantedBy = multi-user.target
```

start the timer, wait for service unit to be activated at the $interval mark. service unit will not be activated once $interval has elapsed unless the timer unit is restarted during the chef converge.

you can see from `systemctl list-timers` output whether it will work in the future, an n/a for "LEFT" or "NEXT" means it will not be triggered.

### Expected Result:
service unit is activated every $interval, not just once after $interval from timer unit starting

### Actual Result:
service unit is only activated once after timer unit is activated

**Comment by nathwill** [comment]
#607 should be reverted

**Comment by nathwill** [comment]
we can potentially use  `OnUnitInactiveSec` instead of `OnUnitActiveSec` to address the problem #607 is trying to address with `Type=oneshot` units not reaching an "activated" state... playing around with this now...

**Comment by ngharo** [comment]
@nathwill did you have reliable timer behavior previous to #607 change?

I'm seeing similar issues you're describing in my environment. Although, I've seen them before the change too.

Based on the man page for `systemd.timer`, the `onActiveSec` _should_ work, don't you think?
> OnActiveSec= defines a timer relative to the moment the timer itself is activated.

Looking at this again, the thing that bothers me is the systemd creator, Lennart Poettering, says not to use OnUnitActiveSec on timers for oneshot services but on my Debian 9 workstation, the systemd-tmpfiles-clean.{timer,service} is exactly that. :man_shrugging: Wondering if there is another root cause...

I'll do some unofficial tests on your PR and report back.

**Comment by nathwill** [comment]
@ngharo we have had reliable activation for the most part. the one exception to that is that a few months back, possibly due to a systemd update (we're on CentOS 7), the service unit stopped activating after reboots, and the timer would not trigger until the service unit had been started. we overrode the systemd_unit[chef-client.service] resource to enable it, and thereafter once the initial chef-client.service run was triggered on boot-up, the timer would actually do its thing.

**Comment by ngharo** [comment]
So far your change is looking promising in my environment of ~50 nodes.

**Comment by tas50** [comment]
Released in 11.1.1
